bioptim/limits/phase_transition.py

```diff
@@ -323,7 +327,11 @@ def custom(transition, all_pn, **extra_params):
             """
 
             nlp_pre, nlp_post = all_pn[0].nlp, all_pn[1].nlp
-            val = transition.custom_function(nlp_pre.states, nlp_post.states, **extra_params)
+
+            states_pre = vertcat(*[nlp_pre.states.cx[i] for i in transition.states_pre_idx])
+            states_post = vertcat(*[nlp_post.states.cx[i] for i in transition.states_post_idx])
```

Comment:
Since this seems to be done quite often, I suggest to create augment the method `__getitem__` of OptimizationVariableList that accepts a list, tuple or slice and perform these actions (so one could do: 
`nlp_pre.states[transition.states_pre_idx].cx` 
instead of manually extracting everything each time